cluster_preflight_check/main.py

```diff
@@ -349,6 +349,8 @@ def setup_logging(context):
 
     # setting handler for logfile
     context.logfile = "/var/log/{}.log".format(context.name)
+    if not utils.is_root():
+        return
```

Comment:
Currently, you check for root on two places: here and in line 366. I would suggest to avoid that.

Furthermore, I'm not sure why you need to store the logger instance in the `context.logger` variable. The logger instance is a so called [*singleton*](https://en.wikipedia.org/wiki/Singleton_pattern), which means, you get always the correct instance regardless where you are (inside a function or module)---as long as you use the correct `logger.getLogger(NAME)` call.

There are probably plenty ways of initializing the logger. See also issue #12, which would probably eliminate the need for the `setup_logging` function.

```python
# inside the main.py file
# skipped lots of imports, but
import logging
from logging.config import fileConfig

# Setup logging, see also issue #12 
fileConfig('your_logging_config.ini')
log = logger.getLogger('cpc')

# ...
# In your function, use log.debug('...') etc. as you know it
```

Reply:
Thanks @tomschr for your great suggestion! I will try to change it!